src/plugins/intel_cpu/src/nodes/scaled_attn.cpp

```diff
@@ -0,0 +1,806 @@
+    MHA_1Token<T> kernel_1tok;
+
+    PlainTensor<T> m_query_emb;  // query with RoPE position embedding
+
+    void prepare_attn_mask(ScaledDotProductAttention* node) {
+        auto memory = node->getParentEdgeAt(3)->getMemoryPtr();
+        attn_mask.resize(memory->getStaticDims());
+        auto p = reinterpret_cast<uint8_t*>(node->getParentEdgeAt(3)->getMemoryPtr()->getData());
+        for (size_t i = 0; i < memory->getSize(); i++)
+            attn_mask.data()[i] = p[i] ? 0.0f : -FLT_MAX;
+    }
+
+    void execute(dnnl::stream strm, ScaledDotProductAttention* node) override {
+        const auto& config = node->get_config();
+        bool has_out_transpose = config.output_BLHxS;
+        bool fuse_causal_attn = config.fuse_causal_attn;
+        bool is_causal = config.is_causal;
+        auto input_num = node->getOriginalInputsNumber();
+        if (fuse_causal_attn && input_num <= 3)
+            IE_THROW() << node->getTypeStr() << " node with name '" << node->getName() << " no attn mask input";
+
+        q_input.reset(node->getParentEdgeAt(0)->getMemoryPtr());
+        k_input.reset(node->getParentEdgeAt(1)->getMemoryPtr());
+        v_input.reset(node->getParentEdgeAt(2)->getMemoryPtr());
+        if (input_num > 3) {
+            // attn_mask
+            if (node->getOriginalInputPrecisionAtPort(3) == Precision::U8) {
+                // bool->f32
+                prepare_attn_mask(node);
+            } else {
+                attn_mask.reset(node->getParentEdgeAt(3)->getMemoryPtr());
+            }
+        }
+        if (input_num > 4) {
+            scale_input = *reinterpret_cast<float*>(node->getParentEdgeAt(4)->getMemoryPtr()->getData());
+        }
+
+        size_t B, H, L1, L0, S;
+
+        // q, k, v: [B, H, L1, S]
+        B = q_input.size(0);
+        H = q_input.size(1);
+        L1 = q_input.size(2);
+        L0 = k_input.size(2) - L1;
+        S = q_input.size(-1);
+
+        if (has_out_transpose)
+            node->redefineOutputMemory({{B, L1, H * S}});
+        else
+            node->redefineOutputMemory({{B, H, L1, S}});
+
+        ov::intel_cpu::PlainTensor<T> output_emb(node->getChildEdgeAt(0)->getMemoryPtr());
+        PlainTensor<T> present_key, present_value;
+
+        q_input.assert_dims({B, H, L1, S});
+        k_input.assert_dims({B, H, L0 + L1, S});
+        v_input.assert_dims({B, H, L0 + L1, S});
+        m_query_emb = q_input;
+        present_key = k_input;
+        present_value = v_input;
+
+        // if fuse_causal_attn == true:
+        //   attn_mask will not be nullptr
+        //   attn_mask will be [B, 1, L1, L0 + L1]
+        //   causal_mask will be got automatically
+        // else:
+        //   if is_causal:
+        //      attn_mask should be nullptr
+        //      causal_mask will be got automatically
+        //   else:
+        //      attn_mask will not be nullptr
+        //      attn_mask will be [B, 1, 1, L0 + L1]
+        //      causal_mask will be nullptr
```

Comment:
these comments seems to be redundant, because the following code is equally clear

Reply:
Removed.